### PR TITLE
[Enhancement] compatible with lower version kudu-client api

### PR DESCRIPTION
**Pull request by predator4ann**
## What I'm doing:

1. Ensures compatibility with older versions of the `kudu-client` API by returning a default row count if the GetTableStatistics method is unsupported.
2. Prioritizes the use of `kudu.table_name` from the metastore when `catalog.type` is configured to `hive`.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/catalog/KuduTable.java [hunk 1]
```diff
@@ -38,31 +38,35 @@ public class KuduTable extends Table {
...
         this.catalogName = catalogName;
         this.databaseName = dbName;
         this.tableName = tblName;
+        this.kuduTableName = kuduTableName;
         this.partColNames = partColNames;
     }
 
     public static KuduTable fromMetastoreTable(org.apache.hadoop.hive.metastore.api.Table table, String catalogName,
                                                List<Column> fullSchema, List<String> partColNames) {
+        String kuduTableName = table.getParameters().get(PARAMETER_KEY_KUDU_TABLE_NAME);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/KuduTable.java:67]: Do you need to check null?

File: fe/fe-core/src/main/java/com/starrocks/connector/kudu/KuduMetadata.java [hunk 2]
```diff
@@ -315,13 +321,24 @@ public Statistics getTableStatistics(OptimizerContext session,
...
-                throw new RuntimeException(e);
+                throw new RuntimeException("Error while accessing Kudu table", e);
             }
         }
         builder.setOutputRowCount(rowCount);
         return builder.build();
     }
+
+    private static boolean isGetTableStatisticsUnsupported(RpcRemoteException e) {
+        return e.getMessage().contains("invalid method name: GetTableStatistics");
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/kudu/KuduMetadata.java:342]: it's a little trick. is there a better way to check this?

Comment (predator4ann) [reply fe/fe-core/src/main/java/com/starrocks/connector/kudu/KuduMetadata.java:342]: I'm afraid no, kuduClient.isStatisticsEnabled() refers to a client-side configuration.

File: fe/fe-core/src/main/java/com/starrocks/catalog/KuduTable.java (hunk 1 above)
Comment (predator4ann) [reply fe/fe-core/src/main/java/com/starrocks/catalog/KuduTable.java:68]: Done.

**Comment by predator4ann** [comment]
@stephen-shelby PTAL ☕

**Review by stephen-shelby — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**